libcaf_core/src/logger.cpp

```diff
@@ -642,12 +643,16 @@ void logger::start() {
     open_file();
     log_first_line();
   } else {
-    thread_ = std::thread{[this] {
+    // Note: we don't call system_->launch_thread here since we don't want to
+    //       set a logger context in the logger thread.
+    auto f = [this](auto guard) {
+      CAF_IGNORE_UNUSED(guard);
       detail::set_thread_name("caf.logger");
       this->system_.thread_started();
       this->run();
       this->system_.thread_terminates();
```

Comment:
```suggestion
      system_.thread_started();
      run();
      system_.thread_terminates();
```

Those seem unnecessary.

Reply:
Yep. Fixed.